src/main/java/org/spongepowered/api/network/ChannelBuf.java

```diff
@@ -493,22 +691,76 @@
     ChannelBuf setString(int index, String data);
 
     /**
-     * Gets a string at the current readerIndex and increases the
-     * readerIndex by 1 in this buffer.
+     * Gets a string at the current readerIndex and increases the readerIndex by
+     * the length of a varint followed by the utf-8 bytes of the string.
+     * 
+     * <p>The string will be encoded as the utf length as a varint followed by
+     * the UTF-8 bytes of the string.</p>
      *
      * @return The string
      */
     String readString();
 
     /**
      * Gets a string at the specified absolute index in this buffer.
+     * 
+     * <p>The string will be encoded as the utf length as a varint followed by
+     * the UTF-8 bytes of the string.</p>
      *
      * @param index The index
      * @return The string
      */
     String getString(int index);
 
     /**
+     * Sets the specified string at the current writerIndex and increases the
+     * writerIndex by the length of the string.
+     * 
+     * <p>The string will be encoded as the utf length as a short followed by
+     * the UTF-8 bytes of the string.</p>
+     *
+     * @param data The boolean data
+     * @return This stream for chaining
+     */
+    ChannelBuf writeUTF(String data);
```

Comment:
as documented in the javadoc the difference is the way the length is encoded, mojang uses a varint in its packets, other things using javas DataOutput/InputStream expect a short.